src/text.rs

```diff
@@ -434,65 +427,108 @@ impl TextBox {
+                        if text
+                            .get(cursor.index..cursor.index)?
+                            .contains(|c: char| c.is_whitespace())
                         {
-                            let x = screen_position.0 + highlight_x;
-                            rects.push(Rect::from_min_max(
-                                (x.floor(), y),
-                                ((x + highlight_w).ceil(), y + line_height),
-                            ));
+                            return None;
                         }
-                    }
 
-                    // See https://docs.rs/cosmic-text/0.8.0/cosmic_text/struct.LayoutRun.html#method.highlight implementation
-                    for glyph in line.glyphs.iter() {
-                        let left_glyph_cursor = if line.rtl {
-                            Cursor::new_with_affinity(line.line_i, glyph.end, Affinity::Before)
-                        } else {
-                            Cursor::new_with_affinity(line.line_i, glyph.start, Affinity::After)
-                        };
-                        let right_glyph_cursor = if line.rtl {
-                            Cursor::new_with_affinity(line.line_i, glyph.start, Affinity::After)
-                        } else {
-                            Cursor::new_with_affinity(line.line_i, glyph.end, Affinity::Before)
-                        };
-                        if (left_glyph_cursor >= start_cursor && left_glyph_cursor <= end_cursor)
-                            && (right_glyph_cursor >= start_cursor
-                                && right_glyph_cursor <= end_cursor)
-                        {
-                            selected_text.push_str(&line.text[glyph.start..glyph.end]);
-                        }
-                    }
-                    if select_end.1 > y + line_height {
-                        selected_text.push(' ')
+                        let end_text = text
+                            .get(cursor.index..)
+                            .and_then(|str| str.split_whitespace().next())?;
+                        let end_index = end_text.len() + cursor.index;
+
+                        let start_text = text
+                            .get(..cursor.index)
+                            .and_then(|str| str.split_whitespace().next_back())?;
+                        let start_index = cursor.index - start_text.len();
+
+                        let start = Cursor::new(cursor.line, start_index);
+                        let end = Cursor::new(cursor.line, end_index);
+
+                        (start, end, position.1, position.1)
+                    } // TODO implement
+                    SelectionMode::Line => {
+                        let start = Cursor::new(cursor.line, 0);
+                        let end = Cursor::new(cursor.line, line.text().len());
+                        (start, end, position.1, position.1)
+                    } // TODO implement
```

Comment:
Left-over comments, or is there still more left to handle?

Reply:
Oops